### DISCUSSION TITLE
#7 📑 Weekly Report 7

### DISCUSSION THREAD

**Discussion by fehlend0** [SE-TINF22B6/Statify#62] [Blog]
Hello everyone👋

Since the week is almost over, here are the updates on the project.

## Nonfunctional requirements:

This week's focus was on discussing and documenting nonfunctional requirements, laying the groundwork for future implementation. The scenarios and utility tree (which you can see below) provide a comprehensive overview of the quality attributes to be addressed in the upcoming development phases. As we move forward, the next steps will involve translating these documented requirements into actionable tasks for implementation and refinement.

#### Quality Attribute Scenarios

| Scenario | Quality Attribute       | Who/What              | Event                     | Influence                | Condition                  | Action                                          | Measurement                                                                 |
|-----------------------------|-----------------------------|-----------------------|---------------------------|---------------------------|----------------------------|-------------------------------------------------|-----------------------------------------------------------------------------|
| 1 | Security (Confidentiality)   | User                  | wishes to connect to Spotify account | Data within the system | Normal operation          | successful connection                          | authentication success                                                      |
| 2 | Usability (Feedback)         | User                  | inputs data               | Data within the system | Normal operation          | gets response if data is valid                  | show error message when data is invalid                                     |
| 3 | Security (Availability)      | Application           | refreshes the access token | Data within the system | Runtime                   | the application is able to continue accessing Spotify data | every 60 mins                                                                           |
| 4 | Usability (Compliance)       | Designer/Developer    | designs the UI             | UI                      | Design time               | the UI fulfills Spotify design guidelines      | [Spotify Design Guidelines](https://developer.spotify.com/documentation/design) |
| 5 | Performance (Latency)        | User                  | creates an account         | System                   | Runtime                   | user can generate statistics    | within 5 mins                                                                           |
| 6 | Performance (Latency)        | Developer             | develops data processing   | Code                    | Development               | data processing should be efficient            | generated data is available in 1 min                                        |
| 7 | Modifiability (Loose coupling, high cohesion) | Developer | develops modules and components | Code               | Development               | reduced interdependence of the components     | changes can be made easily                                                  |
| 8 | Usability (Customizability)  | User                  | wishes to use the application on different devices | UI | Normal operation | the design is responsive                      | all functions are available on any device                                   |
| 9 | Availability (Spotify API issues) | Spotify API       | is down                    | System                   | Normal operation          | log error message                              | the application responds, but only the functions not related to the Spotify API can be used |
| 10 | Performance (Latency)        | Frontend              | sends an API request to the backend | System            | Normal operation          | receives response                     | in 2 sec                                                                           |

#### Utility tree

| Quality attribute    | Refinement             | Quality attribute scenarios   | Business value | Technical risk  |
| :---                 | :----                  | :----                         | :----          | :----           | 
| Security    | Confidentiality         | Scenario 1  (see the table above)                |  H        | L         |
|  Performance | Latency | Scenario 6 (see the table above)                  |  H        | H         |
| Modifiability | loose coupling, high cohesion    | Scenario 7 (see the table above)                 |  M        | H         |

We're looking forward to the next steps in our project. If you have any questions or need further information, feel free to reach out. Let's keep up the good work!

Your Statify Team

**Comment by BenjaminB109** [comment]
Dear Statify Team,
The focus on nonfunctional requirements, especially the detailed Quality Attribute Scenarios and Utility Tree, sets a solid foundation.  Keep up the good work!
